[llvm] Cache llvm runtime bitcode to reduce start up time in dev mode (#1144)
taichi-dev/taichi: docs/global_settings.rst

Global settings
---------------

Backends
********

- To specify which Arch to use: ``export TI_ARCH=cuda`` or ``ti.init(arch=ti.cuda)``.
- To specify which GPU to use for CUDA: ``export CUDA_VISIBLE_DEVICES=[gpuid]``.
- To specify pre-allocated memory size for CUDA: ``export TI_DEVICE_MEMORY_GB=0.5`` or ``ti.init(device_memory_GB=0.5)``.

> - To not use unified memory for CUDA: ``export TI_USE_UNIFIED_MEMORY=0``.

Reviewer: ```suggestion
- To disable unified memory usage on CUDA: ``export TI_USE_UNIFIED_MEMORY=0``.
```

Compilation
***********

- Disable advanced optimization to save compile time & possible erros: ``ti.core.toggle_advanced_optimization(False)``.
- To print intermediate IR generated: ``export TI_PRINT_IR=1`` or ``ti.init(print_ir=True)``.

> - To print preprocessed Python codes: ``export TI_PRINT_PREPROCESSED=1`` or ``ti.init(print_preprocessed=True)``..

Reviewer: ```suggestion
- To print preprocessed Python code: ``export TI_PRINT_PREPROCESSED=1`` or ``ti.init(print_preprocessed=True)``..
```

Runtime
*******

> - Restart the Taichi runtime system (clear memory, destroy all variables and kernels): ``ti.reset()``.

Reviewer: ```suggestion
- Restart the entire Taichi system (destroy all tensors and kernels): ``ti.reset()``.
```
